transaction/std/src/input_credentials.rs

```diff
@@ -78,21 +80,51 @@ impl InputCredentials {
             .position(|element| *element == real_input)
             .expect("Must still contain real input");
 
+        let masked_amount = &ring[real_index].masked_amount;
+        let (amount, blinding) = masked_amount.get_value(&tx_out_shared_secret)?;
+
+        let input_secret = InputSecret {
+            onetime_private_key,
+            amount,
+            blinding,
+        };
+
         Ok(InputCredentials {
             ring,
             membership_proofs,
             real_index,
-            onetime_private_key,
-            real_output_public_key,
-            view_private_key,
+            input_secret,
         })
     }
 }
 
+impl From<InputCredentials> for SignableInputRing {
+    fn from(src: InputCredentials) -> SignableInputRing {
+        SignableInputRing {
+            members: src
+                .ring
+                .iter()
+                .map(|tx_out| (tx_out.target_key, tx_out.masked_amount.commitment))
+                .collect(),
+            real_input_index: src.real_index,
+            input_secret: src.input_secret.clone(),
+        }
+    }
+}
+
+impl From<&InputCredentials> for TxIn {
+    fn from(input_credential: &InputCredentials) -> TxIn {
+        TxIn {
+            ring: input_credential.ring.clone(),
+            proofs: input_credential.membership_proofs.clone(),
+            input_rules: None,
+        }
+    }
+}
+
 impl Zeroize for InputCredentials {
     fn zeroize(&mut self) {
-        self.onetime_private_key.zeroize();
-        self.view_private_key.zeroize();
+        self.real_index.zeroize();
```

Comment:
Shouldn't `input_secret` be zeroized as well?

Reply:
yeah probably all of this should be zeroized, there was some build issue when i was looking at it though, so i skipped it. will look at it again